roles/pulp_common/tasks/install_packages.yml

```diff
@@ -1,4 +1,31 @@
 ---
+- block:
+    - name: "Enumerate the list of updates from the {{ __pulp_pkgs_repo_name }} repo"
+      yum:
+        list: updates
+        disablerepo: "*"
+        enablerepo:  "{{ __pulp_pkgs_repo_name }}"
+      register: updates
+
+    - name: List packages to be upgraded
+      debug:
+        # We do not use json_query because it requires jmespath on the control
+        # node. We should not introduce an addtl control node dependency
+        # unless there is a very compelling reason.
+        # msg: "{{ updates | json_query('results[*].name') }}"
+        msg: "{{ updates.results | map(attribute='name')|list }}"
+
+    - name: Upgrade all existing installed Pulp packages
+      yum:
+        name: "{{ updates.results | map(attribute='name')|list }}"
+        state: latest # noqa 403
+
+  become: true
+  when:
+    - ansible_facts.os_family == "RedHat"
+    - pulp_pkg_repo is not none
+    - pulp_pkg_upgrade_all
```

Comment:
cool!

Reply:
@fao89  You find my when condition impressive? I thought it wasn't particularly ;)

Reply:
I like the `pulp_pkg_upgrade_all`